src/frontend/progress_bar.rs

```diff
@@ -0,0 +1,125 @@
+use gtk::prelude::*;
+use relm4::prelude::*;
+use std::cell::RefCell;
+use std::f64::consts::PI;
+use std::rc::Rc;
+
+pub(crate) const DEFAULT_TOOLTIP_ETA_PROGRESS_BAR: &str = "ETA for next reward payment";
+
+#[derive(Debug, Clone)]
+pub struct CircularProgressBar {
+    progress: Rc<RefCell<f64>>,
+    tooltip_text: String,
+    diameter: f64,
+}
+
+#[derive(Debug)]
+pub enum ProgressBarInput {
```

Comment:
```suggestion
pub enum CircularProgressBarInput {
```

And move it before `CircularProgressBar`. It is inconvenient to read code when there are extra things between struct definition and its implementations. All other examples in the repo are following that pattern.